chalk-ir/src/lib.rs

```diff
@@ -146,6 +193,12 @@ impl UniverseIndex {
 #[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
 pub struct StructId<I: Interner>(pub I::DefId);
 
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct FnDefId<I: Interner>(pub I::DefId);
+
+#[derive(Copy, Clone, PartialEq, Eq, PartialOrd, Ord, Hash)]
+pub struct ClosureId<I: Interner>(pub I::DefId);
```

Comment:
We had previously talked about moving all of these as associated types on `Interner`. This is making me somewhat think this is a good idea. But maybe not in this PR.